examples/decoding/plot_decoding_ssd_csp_eeg.py

```diff
@@ -0,0 +1,256 @@
+# ^^^^^^^^^^^^^^^^^^^^^^^^
+
+# copy raw data
+raw_ssd_transformed = raw_ssd.copy()
+# define SSD filter parameters
+filt_params_signal = dict(l_freq=freq_signal[0], h_freq=freq_signal[1],
+                          l_trans_bandwidth=4, h_trans_bandwidth=4)
+filt_params_noise = dict(l_freq=freq_noise[0], h_freq=freq_noise[1],
+                         l_trans_bandwidth=4, h_trans_bandwidth=4)
+
+# Since we are working with a high electrode counting dataset (64 channels),
+# it is interesting to see the impact of the number of component in SSD, i.e.,
+# the impact in data dimensionality reduction before applying CSP.
+# The minimum n_components is 4, since 4 CSP components were selected
+# the maximum is the number of channels, here 64.
+steps = 4
+n_components = np.arange(4, 65, steps)
+scores_ssd_csp = np.zeros((len(n_components,)))
+std_ssd_csp = np.zeros((len(n_components,)))
+for n, n_comp in enumerate(n_components):
+    print('RUNNING n_components_' + str(n_comp))
+    # set to true return_filtered so as the transformed signal will be filtered
+    # at the desired frequency band.
+    ssd = SSD(raw_ssd.info, filt_params_signal, filt_params_noise,
+              sort_by_spectral_ratio=True, return_filtered=True,
+              n_components=n_comp)
+    # fit and transform
+    raw_ssd_transformed._data = ssd.fit_transform(raw_ssd.get_data())
+    # Now extract epochs between 1 and 2s from the transformed signals
+    epochs_ssd = Epochs(raw_ssd_transformed, events, event_id, tmin, tmax,
+                        proj=True, picks=np.arange(0, n_comp, 1),
+                        baseline=None, preload=True)
+    epochs_ssd_cropped = epochs_ssd.copy().crop(tmin=1., tmax=2.)
+    labels = epochs_ssd.events[:, -1] - 2
+    epochs_ssd_data = epochs_ssd_cropped.get_data()
+    scores_ssd_csp[n] = cross_val_score(pipe_csp, epochs_ssd_data, labels,
+                                        cv=cv, n_jobs=1).mean()
+    std_ssd_csp[n] = cross_val_score(pipe_csp, epochs_ssd_data, labels, cv=cv,
+                                     n_jobs=1).std()
+###############################################################################
+# Let's visualize the results
+fig, ax = plt.subplots(1, dpi=150)
+x_axis = np.arange(0, len(n_components), 1)
+ax.plot(x_axis, scores_ssd_csp, label='SSD+CSP')
+ax.fill_between(x_axis, scores_ssd_csp - std_ssd_csp,
+                scores_ssd_csp + std_ssd_csp, alpha=0.3)
+ax.yaxis.grid(True)
+ax.axhline(scores_csp, linestyle='-', color='k', label='CSP')
+ax.set_xlabel('number of components SSD')
+ax.set_ylabel('classification accuracy')
+ax.set_xticks(np.arange(0, len(n_components), 2))
+ax.set_xticklabels(np.arange(4, 65, steps * 2))
+ax.set_title('Impact of SSD before epoching')
+
+ax.legend()
+# The results show how SSD can be used to improve decoding performance.
+# It is clear that there is a maximum achieved accuracy value, and then
+# the performance of the decoding model decreases as the n_components increases
```

Comment:
This can maybe be described/explained a bit better. When is the maximum value achieved and why?